### PR TITLE
Add Promise Prototype Dispatcher

### PR DESCRIPTION
**Pull request by Yuyupo**
JerryScript-DCO-1.0-Signed-off-by: Daniella Barsony bella@inf.u-szeged.hu

### CODE REVIEW

File: jerry-core/ecma/builtin-objects/ecma-builtin-promise-prototype.c [hunk 1]
```diff
@@ -36,54 +52,42 @@
...
-ecma_builtin_promise_prototype_finally (ecma_value_t this_arg, /**< this argument */
-                                        ecma_value_t on_finally) /**< on_finally function */
-{
-  return ecma_promise_finally (this_arg, on_finally);
-} /* ecma_builtin_promise_prototype_finally */
+  switch (builtin_routine_id)
+  {
+    case ECMA_PROMISE_PROTOTYPE_ROUTINE_THEN:
+    {
+      return ecma_promise_then (this_arg, arguments_list_p[0], arguments_list_p[1]);
```
Comment (rerobika) [inline jerry-core/ecma/builtin-objects/ecma-builtin-promise-prototype.c:74]: Accessing directly args_p[0] is unsafe. Check argc first.

**Review by rerobika — CHANGES_REQUESTED** [review]

File: jerry-core/ecma/builtin-objects/ecma-builtin-promise-prototype.c [hunk 2]
```diff
@@ -36,54 +52,42 @@
...
-ecma_builtin_promise_prototype_finally (ecma_value_t this_arg, /**< this argument */
-                                        ecma_value_t on_finally) /**< on_finally function */
-{
-  return ecma_promise_finally (this_arg, on_finally);
-} /* ecma_builtin_promise_prototype_finally */
+  switch (builtin_routine_id)
+  {
+    case ECMA_PROMISE_PROTOTYPE_ROUTINE_THEN:
+    {
+      return ecma_promise_then (this_arg, argument, arguments_list_p[1]);
```
Comment (rerobika) [inline jerry-core/ecma/builtin-objects/ecma-builtin-promise-prototype.c:74]: ditto, accessing `arguments_list_p[1]` directly is unsafe

**Review by rerobika — CHANGES_REQUESTED** [review]

**Review by rerobika — APPROVED** [review]
LGTM

**Review by zherczeg — APPROVED** [review]
LGTM

**Pull request merged**